stellarphot/core.py

```diff
@@ -715,6 +804,317 @@ def catalog_name(self):
+        # element.
+        cat = cat[0]
+
+        if prepare_catalog is not None:
+            final_cat = prepare_catalog(cat)
+        else:
+            final_cat = CatalogData._tidy_vizier_catalog(cat, mag_column_regex,
+                                                         color_column_regex)
+
+        # Since we go through pandas, we lose the units, so we need to add them back
+        # in.
+        #
+        # We need to swap the key/values on the input map to get the old column names
+        # as values.
+        invert_map = {v: k for k, v in colname_map.items()}
+        final_cat[invert_map['ra']].unit = u.deg
+        final_cat[invert_map['dec']].unit = u.deg
+
+        # Make the CatalogData object....
+        cat = cls(input_data=final_cat,
+                  colname_map=colname_map,
+                  catalog_name=desired_catalog,
+                  catalog_source='Vizier')
+
+        # ...and now that the column names are standardized, clip by frame if
+        # desired.
+        if clip_by_frame:
+            cat_coords = SkyCoord(ra=cat['ra'], dec=cat['dec'])
+            wcs = WCS(header_or_center)
+            x, y = wcs.all_world2pix(cat_coords.ra, cat_coords.dec, 0)
+            in_x = (x >= padding) & (x <= wcs.pixel_shape[0] - padding)
+            in_y = (y >= padding) & (y <= wcs.pixel_shape[1] - padding)
+            in_fov = in_x & in_y
+            cat = cat[in_fov]
+
+        return cat
+
+
+def apass_dr9(header_or_center,
+              radius=1 * u.degree,
+              clip_by_frame=False,
+              padding=100):
+    """
+    Return the items from APASS DR9 that are within the search radius and
+    (optionally) within the field of view of a frame.
+
+    Parameters
+    ----------
+    header_or_center : FITS header or `astropy.coordinates.SkyCoord`
+        Either a FITS header with WCS information or a `SkyCoord` object.
+        The center of the frame or the input coordinate is the center
+        of the cone search.
+
+    radius : `astropy.units.Quantity`, optional
+        Radius around which to search.
+
+    clip_by_frame : bool, optional
+        If ``True``, only return items that are within the field of view
+        of the frame. Default is ``True``.
+
+    padding : int, optional
+        Coordinates need to be at least this many pixels in from the edge
+        of the frame to be considered in the field of view. Default value
+        is 100.
+
+    """
+    apass_colnames = {
+        'recno': 'id', # There is no APASS ID, this is the one generated by Vizier
+        'RAJ2000': 'ra',
+        'DEJ2000': 'dec',
+    }
+    return CatalogData.from_vizier(header_or_center,
+                                   'II/336/apass9',
+                                   radius=radius,
+                                   clip_by_frame=clip_by_frame,
+                                   padding=padding,
+                                   colname_map=apass_colnames)
```

Comment:
Given this construction, I can see why you added this function to core, but it really should probably go a separate file, I think... at least based on it not really being core functionality.  Same for `vsx_vizier`